procrastinate/worker.py

```diff
@@ -102,8 +106,14 @@ async def single_worker(self):
                     },
                 )
                 self.notify_event.clear()
-                await asyncio.wait([self.notify_event.wait()], timeout=self.timeout)
-                self.notify_event.clear()
+                try:
+                    await asyncio.wait_for(
+                        self.notify_event.wait(), timeout=self.timeout
+                    )
+                    self.notify_event.clear()
```

Comment:
Any reason for calling `clear` within the `try` block?

Reply:
It's either this or in the `else`, because if we go through `except`, there's no need to clear because the event wasn't set.

So we could:
- Leave as-is
- Move it to `else` (but it's a bit less readable, but more correct)
- Move it out of the `try` but then we'll clear even when there's a timeout, which is useless

Reply:
My preference is either "move it to `else`" or "move it out of the `try`". I don't like "leave as-is" too much, but reading the code you don't quite know what can cause the `TimeoutError` exception (although it's quite clear in that particular case).